compiler/resolution/implementForallIntents.cpp

```diff
@@ -1723,6 +1724,105 @@ void implementForallIntents2wrapper(CallExpr* call, CallExpr* eflopiHelper)
 // for the (new) ForallStmt-based representation.
 //
 
+// Forall Intent information about each outer variable.
+class FIinfo {
+public:
+  Symbol* ovar;       // outer variable
+  Symbol* svar;       // shadow variable
+  Symbol* reduceGVar; // reduction class, non-NULL only for reduce intents
+  Symbol* reduceMark; // transitional measure; TODO: remove
+  TFITag  intent;     // the forall intent for this variable
+
+  bool isReduce() { return intent == TFI_REDUCE; }
+
+  FIinfo(Symbol* outerVar):
+    ovar(outerVar), svar(NULL),
+    reduceGVar(NULL), reduceMark(NULL), intent(TFI_DEFAULT)
+  { }
+};
```

Comment:
Why is FIinfo local to this file, vs being a full-on AST node with its own header file etc etc?
Also, can we please name it ForallIntentInfo ? Or even just ForallIntent ?